File not found errors in gammapy integration test

Hi, I am trying to use the following script to do an integration test in iminuit for gammapy

```
#!/bin/bash
# run this before a release to check that gammapy is not broken
if [ ! -e gammapy ]; then
  virtualenv gammapy
fi
source gammapy/bin/activate || { echo "Error: You must execute or source this script!"; exit 0; }
pip install cython numpy pytest
pip install .
pip install gammapy pytest-astropy
cd /tmp
gammapy info
python -c "import sys, gammapy; sys.exit(gammapy.test())"
```

but I am getting gammapy-internal errors, such as

```
======================================================================================== ERRORS =========================================================================================
_________________________________________________________________________ ERROR at setup of test_map_maker_ring _________________________________________________________________________

    @pytest.fixture(scope="session")
    def observations():
>       data_store = DataStore.from_dir("$GAMMAPY_DATA/cta-1dc/index/gps/")

/home/hdembinski/Code/iminuit/gammapy/lib/python3.6/site-packages/gammapy/cube/tests/test_make.py:17: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'gammapy.data.data_store.DataStore'>, base_dir = PosixPath('$GAMMAPY_DATA/cta-1dc/index/gps')
hdu_table_filename = PosixPath('$GAMMAPY_DATA/cta-1dc/index/gps/hdu-index.fits.gz'), obs_table_filename = PosixPath('$GAMMAPY_DATA/cta-1dc/index/gps/obs-index.fits.gz')

    @classmethod
    def from_dir(cls, base_dir, hdu_table_filename=None, obs_table_filename=None):
        """Create from a directory.
    
        Parameters
        ----------
        base_dir : str, Path
            Base directory of the data files.
        hdu_table_filename : str, Path
            Filename of the HDU index file. May be specified either relative
            to `base_dir` or as an absolute path. If None, the default filename
            will be looked for.
        obs_table_filename : str, Path
            Filename of the observation index file. May be specified either relative
            to `base_dir` or as an absolute path. If None, the default filename
            will be looked for.
        """
        base_dir = make_path(base_dir)
    
        if hdu_table_filename:
            hdu_table_filename = make_path(hdu_table_filename)
            if (base_dir / hdu_table_filename).exists():
                hdu_table_filename = base_dir / hdu_table_filename
        else:
            hdu_table_filename = base_dir / cls.DEFAULT_HDU_TABLE
    
        if obs_table_filename:
            obs_table_filename = make_path(obs_table_filename)
            if (base_dir / obs_table_filename).exists():
                obs_table_filename = base_dir / obs_table_filename
        else:
            obs_table_filename = base_dir / cls.DEFAULT_OBS_TABLE
    
        if not hdu_table_filename.exists():
>           raise IOError("File not found: {}".format(hdu_table_filename))
E           OSError: File not found: $GAMMAPY_DATA/cta-1dc/index/gps/hdu-index.fits.gz

/home/hdembinski/Code/iminuit/gammapy/lib/python3.6/site-packages/gammapy/data/data_store.py:110: OSError
```

How can I fix this?

--- a/.travis.yml
+++ b/.travis.yml
@@ -27,11 +27,11 @@ env:
         - NUMPY_VERSION=1.15
         - ASTROPY_VERSION=stable
 
-        - CONDA_DEPENDENCIES='Cython click scipy healpy matplotlib pyyaml uncertainties pandas naima sherpa libgfortran regions reproject pandoc ipython iminuit=1.3.3'
-        - CONDA_DEPENDENCIES_OSX='Cython click scipy healpy matplotlib pyyaml uncertainties pandas naima sherpa regions reproject pandoc ipython iminuit=1.3.3'
-        - CONDA_DEPENDENCIES_WO_SHERPA='Cython click scipy healpy matplotlib pyyaml uncertainties pandas naima regions reproject pandoc ipython iminuit=1.3.3'
+        - CONDA_DEPENDENCIES='Cython click scipy healpy matplotlib pyyaml uncertainties pandas naima sherpa libgfortran regions reproject pandoc ipython iminuit'
+        - CONDA_DEPENDENCIES_OSX='Cython click scipy healpy matplotlib pyyaml uncertainties pandas naima sherpa regions reproject pandoc ipython iminuit'
+        - CONDA_DEPENDENCIES_WO_SHERPA='Cython click scipy healpy matplotlib pyyaml uncertainties pandas naima regions reproject pandoc ipython iminuit'
         - CONDA_DOCS_DEPENDENCIES='Cython click scipy healpy matplotlib pyyaml uncertainties pandas naima pygments sherpa libgfortran regions reproject pandoc ipython jupyter'
-        - CONDA_DEPENDENCIES_NOTEBOOKS='Cython click scipy healpy matplotlib pyyaml uncertainties pandas naima sherpa libgfortran iminuit=1.3.3 regions reproject pandoc ipython jupyter'
+        - CONDA_DEPENDENCIES_NOTEBOOKS='Cython click scipy healpy matplotlib pyyaml uncertainties pandas naima sherpa libgfortran iminuit regions reproject pandoc ipython jupyter'
 
         - PIP_DEPENDENCIES='nbsphinx sphinx-click sphinx_rtd_theme pytest-astropy'
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -46,8 +46,6 @@ env:
 
         - ASTROPY_USE_SYSTEM_PYTEST=1
 
-    matrix:
-        - PYTHON_VERSION=3.5 SETUP_CMD='test' FETCH_GAMMAPY_DATA=false
 
 matrix:
 
@@ -73,7 +71,8 @@ matrix:
 
         # Run tests without GAMMAPY_EXTRA available
         - os: linux
-          env: FETCH_GAMMAPY_EXTRA=false PYTHON_VERSION=3.7 SETUP_CMD='test -V'
+          env: FETCH_GAMMAPY_DATA=false PYTHON_VERSION=3.7 SETUP_CMD='test -V'
+               FETCH_GAMMAPY_EXTRA=false
                CONDA_DEPENDENCIES=$CONDA_DEPENDENCIES
 
         # Build docs

--- a/.travis.yml
+++ b/.travis.yml
@@ -27,11 +27,11 @@ env:
         - NUMPY_VERSION=1.15
         - ASTROPY_VERSION=stable
 
-        - CONDA_DEPENDENCIES='Cython click scipy healpy matplotlib pyyaml uncertainties pandas naima sherpa libgfortran regions reproject pandoc ipython iminuit'
-        - CONDA_DEPENDENCIES_OSX='Cython click scipy healpy matplotlib pyyaml uncertainties pandas naima sherpa regions reproject pandoc ipython iminuit'
-        - CONDA_DEPENDENCIES_WO_SHERPA='Cython click scipy healpy matplotlib pyyaml uncertainties pandas naima regions reproject pandoc ipython iminuit'
+        - CONDA_DEPENDENCIES='Cython click scipy healpy matplotlib pyyaml uncertainties pandas naima sherpa libgfortran regions reproject pandoc ipython iminuit=1.3.3'
+        - CONDA_DEPENDENCIES_OSX='Cython click scipy healpy matplotlib pyyaml uncertainties pandas naima sherpa regions reproject pandoc ipython iminuit=1.3.3'
+        - CONDA_DEPENDENCIES_WO_SHERPA='Cython click scipy healpy matplotlib pyyaml uncertainties pandas naima regions reproject pandoc ipython iminuit=1.3.3'
         - CONDA_DOCS_DEPENDENCIES='Cython click scipy healpy matplotlib pyyaml uncertainties pandas naima pygments sherpa libgfortran regions reproject pandoc ipython jupyter'
-        - CONDA_DEPENDENCIES_NOTEBOOKS='Cython click scipy healpy matplotlib pyyaml uncertainties pandas naima sherpa libgfortran iminuit regions reproject pandoc ipython jupyter'
+        - CONDA_DEPENDENCIES_NOTEBOOKS='Cython click scipy healpy matplotlib pyyaml uncertainties pandas naima sherpa libgfortran iminuit=1.3.3 regions reproject pandoc ipython jupyter'
 
         - PIP_DEPENDENCIES='nbsphinx sphinx-click sphinx_rtd_theme pytest-astropy'
 
@@ -47,7 +47,7 @@ env:
         - ASTROPY_USE_SYSTEM_PYTEST=1
 
     matrix:
-        - PYTHON_VERSION=3.5 SETUP_CMD='egg_info' FETCH_GAMMAPY_DATA=false
+        - PYTHON_VERSION=3.5 SETUP_CMD='test' FETCH_GAMMAPY_DATA=false
 
 matrix:
 

--- a/environment-dev.yml
+++ b/environment-dev.yml
@@ -30,7 +30,7 @@ dependencies:
   - healpy
   - reproject
   - sherpa
-  - iminuit>=1.3.2
+  - iminuit
   - pytest
   - pytest-cov
   - sphinx

--- a/gammapy/astro/darkmatter/tests/test_spectra.py
+++ b/gammapy/astro/darkmatter/tests/test_spectra.py
@@ -16,6 +16,7 @@ def test_primary_flux():
     assert_quantity_allclose(actual, desired)
 
 
+@requires_data("gammapy-data")
 def test_DMAnnihilation():
 
     channel = "b"

--- a/gammapy/cube/tests/test_background.py
+++ b/gammapy/cube/tests/test_background.py
@@ -142,6 +142,7 @@ def test_make_map_background_irf(bkg_3d, pars, fixed_pointing_info):
     assert_allclose(m.data.sum(), pars["sum"], rtol=1e-5)
 
 
+@requires_data("gammapy-data")
 def test_make_map_background_irf_constant(fixed_pointing_info_aligned):
     m = make_map_background_irf_with_symmetry(
         fpi=fixed_pointing_info_aligned, symmetry="constant"
@@ -155,6 +156,7 @@ def test_make_map_background_irf_constant(fixed_pointing_info_aligned):
             assert_allclose(d[:, 1], d[0, 1])
 
 
+@requires_data("gammapy-data")
 def test_make_map_background_irf_sym(fixed_pointing_info_aligned):
     m = make_map_background_irf_with_symmetry(
         fpi=fixed_pointing_info_aligned, symmetry="symmetric"
@@ -164,6 +166,7 @@ def test_make_map_background_irf_sym(fixed_pointing_info_aligned):
         assert_allclose(d[0, 1], d[2, 1], rtol=1e-4)  # Symmetric along lat
 
 
+@requires_data("gammapy-data")
 def test_make_map_background_irf_asym(fixed_pointing_info_aligned):
     m = make_map_background_irf_with_symmetry(
         fpi=fixed_pointing_info_aligned, symmetry="asymmetric"

--- a/gammapy/cube/tests/test_make.py
+++ b/gammapy/cube/tests/test_make.py
@@ -110,6 +110,7 @@ def test_map_maker(pars, observations, keepdims):
     assert_allclose(background.data.sum(), pars["background"], rtol=1e-5)
 
 
+@requires_data("gammapy-data")
 def test_map_maker_ring(observations):
     ring_bkg = RingBackgroundEstimator(r_in="0.5 deg", width="0.4 deg")
     geomd = geom(ebounds=[0.1, 1, 10])

--- a/gammapy/cube/tests/test_simulate.py
+++ b/gammapy/cube/tests/test_simulate.py
@@ -3,6 +3,7 @@ import numpy as np
 from numpy.testing import assert_allclose
 from astropy.coordinates import SkyCoord
 import astropy.units as u
+from ...utils.testing import requires_data
 from ...irf import load_cta_irfs
 from ...maps import WcsGeom, MapAxis
 from ...spectrum.models import PowerLaw
@@ -12,11 +13,11 @@ from ...cube import MapDataset
 from ..simulate import simulate_dataset
 
 
+@requires_data("gammapy-data")
 def test_simulate():
-    filename = (
+    irfs = load_cta_irfs(
         "$GAMMAPY_DATA/cta-1dc/caldb/data/cta/1dc/bcf/South_z20_50h/irf_file.fits"
     )
-    irfs = load_cta_irfs(filename)
 
     # Define sky model to simulate the data
     spatial_model = SkyGaussian(lon_0="0 deg", lat_0="0 deg", sigma="0.2 deg")

--- a/gammapy/spectrum/tests/test_flux_point_estimator.py
+++ b/gammapy/spectrum/tests/test_flux_point_estimator.py
@@ -4,7 +4,7 @@ import numpy as np
 from numpy.testing import assert_allclose
 from astropy import units as u
 from astropy.coordinates import SkyCoord
-from ...utils.testing import requires_dependency
+from ...utils.testing import requires_dependency, requires_data
 from ...irf import EffectiveAreaTable, load_cta_irfs
 from ..models import PowerLaw, ExponentialCutoffPowerLaw
 from ..simulation import SpectrumSimulation
@@ -40,10 +40,9 @@ def create_fpe(model):
 
 
 def simulate_map_dataset():
-    filename = (
+    irfs = load_cta_irfs(
         "$GAMMAPY_DATA/cta-1dc/caldb/data/cta/1dc/bcf/South_z20_50h/irf_file.fits"
     )
-    irfs = load_cta_irfs(filename)
 
     skydir = SkyCoord("0 deg", "0 deg", frame="galactic")
     edges = np.logspace(-1, 2, 15) * u.TeV
@@ -124,9 +123,9 @@ class TestFluxPointsEstimator:
         fp = fpe_ecpl.estimate_flux_point(fpe_ecpl.e_groups[1])
         assert_allclose(fp["norm"], 1, rtol=1e-1)
 
-
     @staticmethod
     @requires_dependency("iminuit")
+    @requires_data("gammapy-data")
     def test_run_map_pwl(fpe_map_pwl):
         fp = fpe_map_pwl.run(steps=["err", "norm-scan", "ts"])
 
@@ -147,6 +146,7 @@ class TestFluxPointsEstimator:
 
     @staticmethod
     @requires_dependency("iminuit")
+    @requires_data("gammapy-data")
     def test_run_map_pwl_reoptimize(fpe_map_pwl_reoptimize):
         fp = fpe_map_pwl_reoptimize.run(steps=["err", "norm-scan", "ts"])
 

--- a/gammapy/spectrum/tests/test_flux_point_estimator.py
+++ b/gammapy/spectrum/tests/test_flux_point_estimator.py
@@ -48,12 +48,16 @@ def simulate_map_dataset():
     edges = np.logspace(-1, 2, 15) * u.TeV
     energy_axis = MapAxis.from_edges(edges=edges, name="energy")
 
-    geom = WcsGeom.create(skydir=skydir, width=(4, 4), binsz=0.1, axes=[energy_axis], coordsys="GAL")
+    geom = WcsGeom.create(
+        skydir=skydir, width=(4, 4), binsz=0.1, axes=[energy_axis], coordsys="GAL"
+    )
 
     gauss = SkyGaussian("0 deg", "0 deg", "0.4 deg", frame="galactic")
     pwl = PowerLaw(amplitude="1e-11 cm-2 s-1 TeV-1")
     skymodel = SkyModel(spatial_model=gauss, spectral_model=pwl, name="source")
-    dataset = simulate_dataset(skymodel=skymodel, geom=geom, pointing=skydir, irfs=irfs, random_state=0)
+    dataset = simulate_dataset(
+        skymodel=skymodel, geom=geom, pointing=skydir, irfs=irfs, random_state=0
+    )
     return dataset
 
 
@@ -61,7 +65,9 @@ def simulate_map_dataset():
 def fpe_map_pwl():
     dataset = simulate_map_dataset()
     e_edges = [0.1, 1, 10, 100] * u.TeV
-    return FluxPointsEstimator(datasets=[dataset], e_edges=e_edges, norm_n_values=3, source="source")
+    return FluxPointsEstimator(
+        datasets=[dataset], e_edges=e_edges, norm_n_values=3, source="source"
+    )
 
 
 @pytest.fixture(scope="session")
@@ -71,7 +77,13 @@ def fpe_map_pwl_reoptimize():
     dataset.parameters["lon_0"].frozen = True
     dataset.parameters["lat_0"].frozen = True
     dataset.parameters["index"].frozen = True
-    return FluxPointsEstimator(datasets=[dataset], e_edges=e_edges, norm_values=[1], reoptimize=True, source="source")
+    return FluxPointsEstimator(
+        datasets=[dataset],
+        e_edges=e_edges,
+        norm_values=[1],
+        reoptimize=True,
+        source="source",
+    )
 
 
 @pytest.fixture(scope="session")
@@ -130,19 +142,19 @@ class TestFluxPointsEstimator:
         fp = fpe_map_pwl.run(steps=["err", "norm-scan", "ts"])
 
         actual = fp.table["norm"].data
-        assert_allclose(actual, [0.97922 , 0.94081 , 1.074426], rtol=1e-3)
+        assert_allclose(actual, [0.97922, 0.94081, 1.074426], rtol=1e-3)
 
         actual = fp.table["norm_err"].data
-        assert_allclose(actual, [0.069967, 0.052631, 0.093025], rtol=1e-3)
+        assert_allclose(actual, [0.069963, 0.052605, 0.09297], rtol=1e-3)
 
         actual = fp.table["sqrt_ts"].data
-        assert_allclose(actual, [16.165811, 27.121425, 22.040969], rtol=1e-3)
+        assert_allclose(actual, [16.165806, 27.121415, 22.04104], rtol=1e-3)
 
         actual = fp.table["norm_scan"][0]
         assert_allclose(actual, [0.2, 1, 5], rtol=1e-3)
 
         actual = fp.table["dloglike_scan"][0] - fp.table["loglike"][0]
-        assert_allclose(actual, [1.536460e+02, 8.756689e-02, 1.883420e+03], rtol=1e-3)
+        assert_allclose(actual, [1.536452e02, 8.762343e-02, 1.883447e03], rtol=1e-3)
 
     @staticmethod
     @requires_dependency("iminuit")

